components/private_channel/browser/private_channel_service.cc

```diff
@@ -0,0 +1,320 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+
+#include "brave/components/private_channel/browser/private_channel_service.h"
+#include "brave/components/private_channel/browser/static_values.h"
+#include "brave/components/private_channel/browser/request_utils.h"
+#include "brave/components/private_channel/client_private_channel.h"
+#include "brave/components/private_channel/utils.h"
+
+#include "base/logging.h"
+#include "extensions/common/url_pattern.h"
+#include "net/base/load_flags.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "chrome/browser/browser_process.h"
```

Comment:
this is also a layering violation. The way to avoid these is to have a factory method in `brave/browser` that passes in the shared url loader

Reply:
I'm opening a separate PR to fix the layering violation on the referrals code. Then rebase and use it to solve this issue.

- [ ] PR to pass factory method to referrals that can be used by private_channels.